azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/util/DeploymentTemplateBuilder.java

```diff
@@ -370,19 +433,54 @@ private void addVirtualMachine() {
               .diagnosticsProfile(diagnosticsProfile)
               .build();
 
+
+      String tagString = StringUtils.join(Lists.newArrayList(tags), ",");
+      if (tagString.isEmpty())
+         tagString = "jclouds";
+      userMetaData.put("tags", tagString);
+
       variables.put("virtualMachineName", name);
       ResourceDefinition virtualMachine = ResourceDefinition.builder()
               .name("[variables('virtualMachineName')]")
               .type("Microsoft.Compute/virtualMachines")
               .location(location)
-              .apiVersion(STORAGE_API_VERSION)
+              .apiVersion("2015-06-15")
               .dependsOn(Arrays.asList("[concat('Microsoft.Storage/storageAccounts/', variables('storageAccountName'))]",
                       "[concat('Microsoft.Network/networkInterfaces/', variables('networkInterfaceCardName'))]"))
-              .tags(ImmutableMap.of("displayName", "VirtualMachine"))
+              .tags(userMetaData)
               .properties(properties)
               .build();
 
       resources.add(virtualMachine);
    }
 
+
+   private ImageReference getImageReference(String publisher, String offer, String sku) {
+      return ImageReference.builder()
+              .publisher(publisher)
+              .offer(offer)
+              .sku(sku)
+              .version("latest")
+              .build();
+
+   }
+
+   private OSDisk getOsDisk(String vhdUri, String imageUri) {
+      OSDisk.Builder builder = OSDisk.builder();
+      builder.name("[variables('osDiskName')]");
+      builder.caching("ReadWrite");
+      builder.createOption("FromImage");
+      builder.vhd(VHD.create(vhdUri));
+
+      if (!imageUri.isEmpty()) {
+         builder.osType("Linux");
```

Comment:
I guess it is not always like that but depending on image it can be Windows as well.